src/Surfnet/StepupGateway/SecondFactorOnlyBundle/Controller/SecondFactorOnlyController.php

```diff
@@ -131,6 +131,13 @@ public function respondAction(Request $request)
 
         // Reset state
         $this->getSecondFactorRespondService()->resetRespondState($responseContext);
+
+        // Handle SAML response
+        $httpResponse =  $responseRendering->renderResponse($responseContext, $response, $request);
+
+        $ssoCookieService = $this->get('gateway.service.sso_2fa_cookie');
+        $ssoCookieService->handleSsoOn2faCookieStorage($responseContext, $request, $httpResponse, 'sfo');
```

Comment:
The writing of the SSO cookie is moved to a point before the ADFS response is sent. The regular response is created at an earlier stage than before, this to be able to create the SSO cookie (it needs the http response object to write the cookie)

This ensures both ADFS and regular SFO authentications can write the sso cookie (when allowed)